### DISCUSSION TITLE
Guide for implementing uniswap v3 simulation using revm

### DISCUSSION THREAD

**Discussion by walkerlala** [bluealloy/revm#1466] [General]
There is an example of implementing uniswap v2 simulation using revm in the revm repo. But uniswap V3 is more complex and involves lots of storage change/watch to be implemented.

Could anyone share a practical example for uniswap v3 ?

**Comment by davebryson** [comment]
@walkerlala  Check out https://github.com/simular-fi/unisim  It uses Simular which is a Python wrapper around revm.  Unisim uses Uniswap v3 contracts and provides an API for creating pools for local experimentation
